evalml/tests/model_understanding_tests/test_graphs.py

```diff
@@ -922,6 +925,30 @@ def test_partial_dependence_errors(logistic_regression_binary_pipeline_class):
         partial_dependence(pipeline, X, features=(0, 'b'))
 
 
+def test_partial_dependence_more_categories_than_grid_resolution(logistic_regression_binary_pipeline_class):
```

Comment:
Just a suggestion but maybe we could somehow make it more clear that the number of categories is greater than the grid resolution, whether that's by setting grid_resolution specifically to a number and then asserting that there are `currency` has more than that number of categories, which is why we trigger this new code? Or alternatively we could have a tiny dataset like the previous test and set grid_resolution to something even tinier?